src/item_location.cpp

```diff
@@ -841,6 +841,11 @@ void item_location::deserialize( const JsonObject &obj )
         item_location parent;
         obj.read( "parent", parent );
         if( !parent.ptr->valid() ) {
+            if( parent.where() == type::invalid ) {
```

Comment:
```suggestion
            if( parent == nowhere ) {
```

Essentially the same check, just simplified.